### PR TITLE
Rtl migration

### PR DESCRIPTION
**Pull request by kawikabader**
## Summary

There have been issues related to newer versions of Nextjs and React with Enzyme.

## Changelog

Replace Enzyme with React Testing Library.

## Test Plan

Updated tests.

### CODE REVIEW

File: packages/gasket-nextjs/package.json [hunk 1]
```diff
@@ -42,20 +42,18 @@
   },
   "homepage": "https://github.com/godaddy/gasket/tree/main/packages/gasket-nextjs",
   "dependencies": {
-    "@gasket/data": "^6.36.1",
+    "@gasket/data": "^6.34.6",
```
Comment (agerard-godaddy) [inline packages/gasket-nextjs/package.json:45]: Is this change intentional? The version should match up to what's current in the monorepo.

**Review by agerard-godaddy — APPROVED** [review]

**Review by mmason2-godaddy — APPROVED** [review]
